portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,31 +14,91 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import com.google.sps.data.Comment;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    //ArrayList<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<String>();
-    comments.add("This website is cool!");
-    comments.add("I like your pictures!");
-    comments.add("The colors look nice together!");
+    Query query = new Query("Comment").addSort("timestamp",SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    //equal string and check to see if it null
+    //if null set equal to default
+    String userNum = request.getParameter("user-comment-num");
+    int numberComments;
+    if (userNum == null){
+        numberComments = 5;
+    }else{
+        numberComments = Integer.parseInt(userNum);
+    }
```

Comment:
nit: Another way to structure this code would be (please use if it looks simpler):

```
    int numberComments = 5;
    if (userNum != null) {
        numberComments = Integer.parseInt(userNum);
    }
```